lib/db/feature-tag-store.js

```diff
@@ -0,0 +1,245 @@
+'use strict';
+
+const metricsHelper = require('../metrics-helper');
+const { DB_TIME } = require('../events');
+const {
+    TAG_CREATED,
+    TAG_DELETED,
+    FEATURE_TAGGED,
+    FEATURE_UNTAGGED,
+} = require('../event-type');
+const { CREATE_TAG, DELETE_TAG } = require('../command-type');
+const NotFoundError = require('../error/notfound-error');
+
+const COLUMNS = ['id', 'type', 'value'];
+const TABLE = 'tags';
+const FEATURE_TAG_TABLE = 'feature_tag';
+
+class FeatureTagStore {
+    constructor(db, eventStore, eventBus, getLogger) {
+        this.db = db;
+        this.eventStore = eventStore;
+        this.logger = getLogger('feature-tag-store.js');
+
+        this.timer = action =>
+            metricsHelper.wrapTimer(eventBus, DB_TIME, {
+                store: 'tag',
+                action,
+            });
+
+        eventStore.on(CREATE_TAG, event => this._createTag(event.data));
+        eventStore.on(DELETE_TAG, event => this._deleteTag(event.data));
+    }
+
+    async getTags() {
+        const timer = this.timer('getTags');
+        const rows = await this.db.select(COLUMNS).from(TABLE);
+        timer();
+        return rows.map(this.rowToTag);
+    }
+
+    async getAllOfType(type) {
+        const timer = this.timer('getAllOfType');
+        const rows = await this.db
+            .select(COLUMNS)
+            .from(TABLE)
+            .where({ type });
+        timer();
+        return rows.map(this.rowToTag);
+    }
+
+    async getAllTagsForFeature(featureName) {
+        const timer = this.timer('getAllForFeature');
+        const rows = await this.db
+            .select('*')
+            .from('features')
+            .join('feature_tag', 'features.name', 'feature_tag.feature_name')
+            .join('tags', 'feature_tag.tag_id', 'tags.id')
+            .where({ name: featureName });
+        timer();
+        return rows.map(this.rowToTag);
+    }
+
+    async getTagByTypeAndValue(type, value) {
+        const stopTimer = this.timer('getTagByTypeAndValue');
+        const row = await this.db
+            .first(COLUMNS)
+            .from(TABLE)
+            .where({ type, value })
+            .then(this.rowToTag);
+        stopTimer();
+        return row;
+    }
+
+    async getTag(id) {
+        const stopTimer = this.timer('getTagById');
+        const tag = this.db
```

Comment:
missing `await`

Reply:
Thanks, good catch, I'll do what you suggested at L94, and return directly, stopping the timer in the `then` clause.